tests/report/reportportal/test.sh

```diff
@@ -1,17 +1,135 @@
 #!/bin/bash
 . /usr/share/beakerlib/beakerlib.sh || exit 1
 
+TOKEN=$TMT_REPORT_REPORTPORTAL_TOKEN
+URL=$TMT_REPORT_REPORTPORTAL_URL
+PROJECT="$(grep 'project' 'data/plan.fmf' | awk -F ': ' '{print $2}')"
+
+# testing all the attributes/parameters under a label
+# $1=rl_message, $2=response, $3=jq_element, $4=fmf_file, $5=fmf_label
+check_attr_label() {
+    [[ $1 ]] && local rl_message=$1
+    [[ $2 ]] && local response=$2
+    [[ $3 ]] && local jq_element=$3
+    [[ $4 ]] && local fmf_file=$4
+    [[ $5 ]] && local fmf_label=$5
+    echo "$response" | jq -r ".$jq_element" > tmp_attributes.json && rlPass "$rl_message" || rlFail "$rl_message"
+    awk -v label=$fmf_label '$0~label,/^$/ {if($0 && $0!=label){print $0}}' $fmf_file | while IFS= read -r line
+        do
+            key="$(echo "$line" | grep -o '[^ :]*' | head -1)"
+            value="$(echo "$line" | grep -o '[^ :]*' | tail -1)"
+            rlAssertGrep "$key" tmp_attributes.json -A1 > tmp_attributes_selection
+            rlAssertGrep "$value" tmp_attributes_selection
+        done
+    rm tmp_attributes*
+}
+
+# testing one of the attributes/parameters under a specific key
+# $1=rl_message, $2=response, $3=jq_element, $4=fmf_file, $5=fmf_key, $6=test_index, $7=inv_grep_key, $8=inv_grep_value
+check_attr_key() {
+    [[ $1 ]] && local rl_message=$1
+    [[ $2 ]] && local response=$2
+    [[ $3 ]] && local jq_element=$3
+    [[ $4 ]] && local fmf_file=$4
+    [[ $5 ]] && local fmf_key=$5
+    [[ $6 ]] && local test_index=$6
+    [[ $7 ]] && local inv_grep_key="Not"   || local inv_grep_key=""
+    [[ $8 ]] && local inv_grep_value="Not" || local inv_grep_value=""
+    echo "$response" | jq -r ".$jq_element" > tmp_attributes.json && rlPass "$rl_message" || rlFail "$rl_message"
+    rlAssert${inv_grep_key}Grep "$fmf_key" tmp_attributes.json -A1 > tmp_attributes_selection
+    if [[ -f $file && ! $inv_grep_value ]]; then
+        value="$(grep $fmf_key $fmf_file | sed -n "$test_index p" | tr -s ' ' | grep -o '[^ :]*' | tail -1)"
+        rlAssert${inv_grep_value}Grep "$value" tmp_attributes_selection
+    fi
+    rm tmp_attributes*
+}
+
+
 rlJournalStart
     rlPhaseStartSetup
         rlRun "pushd data"
         rlRun "run=$(mktemp -d)" 0 "Create run workdir"
+        rlRun "set -o pipefail"
     rlPhaseEnd
 
     rlPhaseStartTest
         rlRun -s "tmt run --id $run --verbose" 2
-        rlAssertGrep "project: tmt" $rlRun_LOG
-        rlAssertGrep "launch: smoke" $rlRun_LOG
-        rlAssertGrep "report: Successfully uploaded." $rlRun_LOG
+        cat  $rlRun_LOG
+        rlAssertGrep "launch: /plan" $rlRun_LOG
+        rlAssertGrep "test: /test/bad" $rlRun_LOG
+        rlAssertGrep "test: /test/good" $rlRun_LOG
+        rlAssertGrep "test: /test/weird" $rlRun_LOG
+        rlAssertGrep "URL: http.*redhat.com.ui/\#${PROJECT}/launches/all/[0-9]{4}" $rlRun_LOG -Eq
+    rlPhaseEnd
+
+    rlPhaseStart FAIL "Interface Test"
```

Comment:
From the document https://beakerlib.readthedocs.io/en/latest/manual.html#rlphasestart, looks `rlPhaseStart FAIL "Interface Test"` can be used `rlPhaseStartTest "Interface Test"` instead as L55 uses `rlPhaseStartTest`. Would you like to update it to make the test code consistent if `rlPhaseStart FAIL ... == rlPhaseStartTest ...` is true?

Reply:
Thanks, updated.